sale_coupon_order_line_link/models/sale_order.py

```diff
@@ -114,29 +93,15 @@ def _link_reward_lines(self, programs):
     def _link_reward_generated_lines(self, programs):
         """Link the lines that generated reward lines to those lines"""
         for program in programs:
-            reward_lines = self.order_line.filtered(
-                lambda x: x.coupon_program_id == program
-            )
+            reward_lines = self.order_line.filtered(lambda x: x.reward_id == program)
             self.order_line._filter_related_program_lines(program).write(
                 {"reward_generated_line_ids": [(4, rl.id) for rl in reward_lines]}
             )
 
-    def _create_new_no_code_promo_reward_lines(self):
-        """Ensure that the links remain"""
-        res = super()._create_new_no_code_promo_reward_lines()
-        for order in self:
-            order._link_reward_generated_lines(order.order_line.coupon_program_id)
-        return res
-
 
 class SaleOrderLine(models.Model):
     _inherit = "sale.order.line"
 
-    coupon_program_id = fields.Many2one(
```

Comment:
Because there is already a field called reward_id in the sale.order.line model that contains the same information. Would it be necessary to duplicate it?

Reply:
But it's not the same record. You can get the data from the reward_id converting it to a related field, but we don't want to miss this information. So:

- Rename the field to loyalty_program_id
- Make it related stored to reward_id